lib/natalie/compiler2/instructions/and_instruction.rb

```diff
@@ -0,0 +1,26 @@
+require_relative './base_instruction'
+
+module Natalie
+  class Compiler2
+    class AndInstruction < BaseInstruction
+      def to_s
+        'and'
+      end
+
+      def generate(transform)
+        option_2 = transform.pop
+        option_1 = transform.pop
+        result = transform.temp("and")
+        transform.push(
+          "(#{option_1}->is_truthy() ? #{option_2} : #{option_1})"
```

Comment:
Unfortunately, this will evaluate `option_1` twice. We'll need to memoize it. Same with `OrInstruction`...